File: LeetCode_Marathons/LCM1/Medium/Two_Sum_II_-_Input_Array_Is_Sorted/a.go
func twoSum(numbers []int, target int) []int {
    n := len(numbers)
    l, r := 0, n - 1;

    for l < r {
        if numbers[l] + numbers[r] == target {
            return []int{l + 1, r + 1}
        } else if numbers[l] + numbers[r] < target {
            l++;
        } else {
            r--;
        }
    }

    return []int{}
}

File: LeetCode_Marathons/LCM1/Medium/Two_Sum_II_-_Input_Array_Is_Sorted/a.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n = numbers.length;

        int l = 0, r = n - 1;
        while (l < r) {
            if (numbers[l] + numbers[r] == target) return new int[] {l + 1, r + 1};
            else if (numbers[l] + numbers[r] < target) l++;
            else r--;
        }

        return new int[] {};
    }
}